### PR TITLE
Music

### PR DESCRIPTION
**Pull request by weeryan17**
# Music PR

This pr is from the music branch into dev. for reference to see what was worked on in this pr see issue #72 .

I'm going to do a final run of music before we merge to try and catch any issues. I'll update this when that's done

### CODE REVIEW

File: src/main/java/org/cascadebot/cascadebot/events/VoiceEventListener.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.commands.music.SkipCommand;
+import org.cascadebot.cascadebot.data.managers.GuildDataManager;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroup;
+
+public class VoiceEventListener extends ListenerAdapter {
+
+    @Override
+    public void onGenericGuildVoice(GenericGuildVoiceEvent event) {
+        // TODO: handle leaving & moving of players so we don't get lonely :(
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/events/VoiceEventListener.java]: This should be completed for this PR

File: src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java [hunk 2]
```diff
@@ -16,14 +17,31 @@
...
     public static final String FAST_FORWARD = "\u23ED";  // ⏭
     public static final String REPEAT = "\uD83D\uDD01"; // 🔁
     public static final String REPEAT_ONCE = "\uD83D\uDD02"; // 🔂
+    public static final String SONG = "\uD83C\uDFB5"; // 🎵
+    public static final String PLAYLIST = "\uD83C\uDFB6"; // 🎶
 
     public static final String FROWNING = "\uD83D\uDE26"; // 😦
     public static final String THINKING = "\uD83E\uDD14"; // 🤔
 
+    public static final String ONE = "\u0031\u20E3";
```
[the remark this answers is not in the document]
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/UnicodeConstants.java]: If i can find the unicode again

File: src/main/java/org/cascadebot/cascadebot/events/VoiceEventListener.java (hunk 1 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/events/VoiceEventListener.java]: Not really as this is more of a beta thing when we allow the bot to stay if people donate

File: src/main/java/org/cascadebot/cascadebot/CascadeBot.java [hunk 3]
```diff
@@ -196,8 +212,20 @@ private void init() {
...
+
     }
 
+    private void setupTasks() {
+        new Task("prune-players") {
+            @Override
+            protected void execute() {
+                musicHandler.purgeDisconnectedPlayers();
+            }
+        }.start(TimeUnit.MINUTES.toMillis(5), TimeUnit.MINUTES.toMillis(15));
```
[the remark this answers is not in the document]
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/CascadeBot.java]: Look below

**Pull request merged**